hr_job_categories/hr.py

```diff
@@ -108,11 +108,11 @@ def write(self, cr, uid, ids, vals, context=None):
         for contract in self.browse(cr, uid, ids, context=context):
             for data in prev_data:
                 if (
-                        data.get('id') == contract.id
-                        and not vals.get('job_id', False)
-                        or (
-                            data.get('job_id')
-                            and data['job_id'][0] != vals.get('job_id')
+                        data.get('id') == contract.id and
```

Comment:
Do we really need spaces between the '(' and the first checked condition ?
IMO the readability can be improved if the code is written as:
```python
            if(data.get('id') == contract.id and
               not vals.get('job_id', False) or
               (data.get('job_id') and
                data['job_id'][0] != vals.get('job_id')
               )
            ):
```